### PR TITLE
[Translation] [russian] - context.article

### PR DESCRIPTION
**Pull request by alexshd**

### CODE REVIEW

File: _content/tour/rus/context.article [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-signals, and other request-scoped values across API boundaries and between processes.
+Пакет `context` определяет тип `Context`, который переносит сроки, отмены
+сигналы и другие значения, связанные с запросами, через границы API и между процессами.
 
-** Code Review
+** Обзор кода
 
-"Context values are for request-scoped data that passes through programs in a
-distributed system.
+"Значения контекста предназначены для данных, копируемых по запросу, которые проходят через программы в
```
Comment (alexshd) [inline _content/tour/rus/context.article:14]: After being stuck on translating this sentence for too long ... Gave it to ChatGPT ...
This word collection has no meaning or is very far from the origin...

HELP Please... :)

File: _content/tour/rus/context.article [hunk 2]
```diff
@@ -1,25 +1,24 @@
-Context Package
-The package context defines the Context type, which carries deadlines, cancellation signals, and other request-scoped values across API boundaries and between processes.
+   Пакет Context
```
Comment (rumyantseva) [inline _content/tour/rus/context.article:1]:
```suggestion
Пакет Context
```

File: _content/tour/rus/context.article [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+Пакет context определяет тип `Context`, который переносит сроки, сигналы отмены и другие значения, связанные с запросами, через границы API и между процессами.
```
Comment (rumyantseva) [inline _content/tour/rus/context.article:2]:
```suggestion
Пакет context определяет тип `Context`, который включает работу со сроками (дедайнами), сигналами отмены и другими значениями, имеющими смысл в рамках запросов API и между процессами.
```

File: _content/tour/rus/context.article [hunk 4]
```diff
@@ -1,25 +1,24 @@
...
-signals, and other request-scoped values across API boundaries and between processes.
+Пакет `context` определяет тип `Context`, который переносит сроки, отмены
+сигналы и другие значения, связанные с запросами, через границы API и между процессами.
 
-** Code Review
+** Обзор кода
 
-"Context values are for request-scoped data that passes through programs in a
-distributed system.
+Значения контекста (`ctx.Value()`) предназначены для данных, скопированных на запрос, которые передаются через программы в распределенной системе.
```
Comment (rumyantseva) [inline _content/tour/rus/context.article:14]:
```suggestion
Значения Context предназначены для работы с данными из области действия запросов передающихся через программы в распределенных системах.
```

**Review by rumyantseva — COMMENTED** [review]
This one is hard. I started reviewing, but I don't really like my suggestions. I need to think more.

File: _content/tour/rus/context.article [hunk 5]
```diff
@@ -156,41 +153,40 @@ a request.
...
-milliseconds. On lines 15-19, the request is created to make a GET call against
-the provided URL. Lines 22-23 create a Context with a 50 millisecond timeout. A
-new API added to the Request value back in version 1.7 is the WithContext method.
-This method allows the Request value’s Context field to be updated. On line 26,
-that is exactly what the code is doing.
+Эта программа отправляет запрос на rss-ленту блога Ardan с тайм-аутом в 50
+миллисекунд. В строках 15-19 запрос создается для выполнения `GET`-вызова по
+предоставленному `URL`. В строках 22-23 создается контекст с таймаутом 50 миллисекунд. A
+Новый `API`, добавленный к значению`Request` еще в версии 1.7, - это метод `WithContext`.
+Этот метод позволяет обновлять поле `Context` значения `Request`. В строке 26,
```
Comment (shalimski) [inline _content/tour/rus/context.article:160]:
```suggestion
Этот метод позволяет обновлять поле `Context` внутри `Request`. В строке 26,
```

File: _content/tour/rus/context.article [hunk 6]
```diff
@@ -156,41 +153,40 @@ a request.
...
+параметр каждого метода это `Context`, и так же соблюдаеться идиома с переменной `ctx`.
 
-*The* *chain* *of* *function* *calls* *between* *them* *must* *propagate* *the* *Context*
+*Цепочка* *вызовов* *функций* *должна* *распространять* *контекст*.
 
-This is an important rule since a Context is request or task based. You want the
-Context and any changes made to it during the processing of the request or task
-to be propagated and respected.
+Это важное правило, поскольку контекст основан на запросах или действиях. Вы хотите, чтобы контекст и любые изменения, 
+сделанные во время обработки запроса или действия, распространялись и соблюдались.
```
Comment (shalimski) [inline _content/tour/rus/context.article:189]:
```suggestion
Это важное правило, поскольку контекст основан на запросах или действиях. Вы хотите, чтобы контекст и любые изменения, внесенные в него во время обработки запроса или задачи, распространялись и соблюдались.
```

File: _content/tour/rus/context.article [hunk 7]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
     33     return web.Respond(ctx, w, users, http.StatusOK)
     34 }
 
-In this listing you see a handler function called List which is executed when a
-user makes an HTTP request for this endpoint. The handler accepts as its first
-parameter a Context, since it’s part of a request and will perform I/O. You can
-see on lines 25, 28 and 33 that the same Context value is propagated down the
-call stack.
+На этом фрагменте кода видно функцию-обработчик с именем `List`,
```
Comment (shalimski) [inline _content/tour/rus/context.article:204]:
```suggestion
В этом фрагменте кода видно функцию с именем `List`,
```

File: _content/tour/rus/context.article [hunk 8]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
+которая выполняется при запросе пользователя через `HTTP` для данной конечной точки.
```
Comment (shalimski) [inline _content/tour/rus/context.article:205]:
```suggestion
которая выполняется, когда пользователь отправляет HTTP-запрос для этой конечной точки.
```

File: _content/tour/rus/context.article [hunk 9]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
+Обработчик принимает контекст (Context) в качестве первого параметра,
```
Comment (shalimski) [inline _content/tour/rus/context.article:206]:
```suggestion
Обработчик принимает контекст `Context` в качестве первого параметра,
```

File: _content/tour/rus/context.article [hunk 10]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
+поскольку он является частью запроса и выполняет ввод-вывод.
```
Comment (shalimski) [inline _content/tour/rus/context.article:207]:
```suggestion
поскольку он является частью запроса и будет выполнять операции ввода-вывода.
```

File: _content/tour/rus/context.article [hunk 11]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
+Как видно на строках 25, 28 и 33, тот же самый контекст (Context) передается вниз по стеку вызовов.
```
Comment (shalimski) [inline _content/tour/rus/context.article:208]:
```suggestion
Как видно на строках 25, 28 и 33, тот же самый контекст `Context` передается вниз по стеку вызовов.
```

File: _content/tour/rus/context.article [hunk 12]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
 
-A new Context value is not created since this function requires no changes to it.
-If a new top-level Context value would be created by this function, any existing
-Context information from a higher-level call associated with this request would
-be lost. This is not what you want.
+Новый объект контекста (Context) не создается, поскольку для этой функции не требуются изменения.
```
Comment (shalimski) [inline _content/tour/rus/context.article:210]:
```suggestion
Новый объект контекста не создается, так как эта функция не требует его изменения.
```

File: _content/tour/rus/context.article [hunk 13]
```diff
@@ -205,16 +201,16 @@ to be propagated and respected.
...
+Если бы новый объект контекста (Context) верхнего уровня был создан этой функцией, 
+любая существующая информация о контексте (Context) из вызова более высокого уровня,
+связанного с этим запросом, была бы утеряна. Это не то, что вам нужно.
```
Comment (shalimski) [inline _content/tour/rus/context.article:213]:
```suggestion
Если бы эта функция создала новое значение контекста верхнего уровня, любая существующая контекстная информация из вызова более высокого уровня, связанная с этим запросом, была бы потеряна. Это не то, что вам нужно.
```

File: _content/tour/rus/context.article [hunk 14]
```diff
@@ -231,18 +227,15 @@ be lost. This is not what you want.
...
     46 }
 
-You see the declaration of the List method that was called on line 28. Once again
-this method accepts a Context as its first parameter. This value is then propagated
-down the call stack once again on lines 35 and 41. Since line 41 is a database call,
-that function should be respecting any timeout information set in the Context from
-any caller above.
+Посмотрите объявление метода `List`, который вызывается в строке 28. И снова этот метод принимает `Context` в качестве первого параметра. 
+Затем это значение снова передается вниз по стеку вызовов в строках 35 и 41. Поскольку строка 41 является вызовом базы данных, 
+эта функция должна уважать любую информацию о тайм-ауте, установленную в контексте предыдущим пользователем.
```
Comment (shalimski) [inline _content/tour/rus/context.article:232]:
```suggestion
эта функция должна учитывать любую информацию о тайм-ауте, установленную в контексте любого вызывающего объекта выше.
```

File: _content/tour/rus/context.article [hunk 15]
```diff
@@ -231,18 +227,15 @@ be lost. This is not what you want.
...
 
-*Replace* *a* *Context* *using* *WithCancel,* *WithDeadline,* *WithTimeout,* *or* *WithValue*
+*Получает* *контекст* *с* *помощью* *WithCancel*, *WithDeadline*, *WithTimeout* *или* *WithValue*.
```
Comment (shalimski) [inline _content/tour/rus/context.article:234]:
```suggestion
*Замена* *контекста* *с* *помощью* *WithCancel*, *WithDeadline*, *WithTimeout* *или* *WithValue*.
```

File: _content/tour/rus/context.article [hunk 16]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
     31     }
     32 }
 
-This is a small program that shows the value semantic nature of the WithTimeout
-function. On line 08, the call to WithTimeout returns a new Context value and a
-cancel function. Since the function call requires a parent Context, the code uses
-the Background function to create a top-level empty Context. This is what the
-Background function is for.
+Это небольшая программа, демонстрирующая семантическую природу значения функции `WithTimeout`.
```
Comment (shalimski) [inline _content/tour/rus/context.article:273]:
```suggestion
Это небольшая программа, которая показывает семантическую природу функции `WithTimeout`. 
```

File: _content/tour/rus/context.article [hunk 17]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
...
+В строке 08 вызов `WithTimeout` возвращает новое значение `Context` и функцию отмены. Поскольку вызов функции требует 
+родительского контекста, код использует функцию `Background` для создания пустого контекста верхнего уровня. 
+В этом и заключается назначение функции Background.
 
-Moving forward the Context value created by the WithTimeout function is used. If
-any future functions in the call chain need their own specific timeout or deadline,
-they should also use the appropriate With function and this new Context value as
-the parent.
+В дальнейшем используется значение `Context`, созданное функцией `WithTimeout`. Если каким-либо последующим функциям в 
+цепочке вызовов требуется определенный таймаут или крайний срок, они также должны использовать соответствующую функцию `With`
```
Comment (shalimski) [inline _content/tour/rus/context.article:279]:
```suggestion
цепочке вызовов потребуется свой собственный определенный таймаут или крайний срок, они также должны использовать соответствующую функцию `With` 
```

File: _content/tour/rus/context.article [hunk 18]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
...
-the parent.
+В дальнейшем используется значение `Context`, созданное функцией `WithTimeout`. Если каким-либо последующим функциям в 
+цепочке вызовов требуется определенный таймаут или крайний срок, они также должны использовать соответствующую функцию `With` 
+и это новое значение Context в качестве родительского.
 
-It’s critically important that any cancel function returned from a With function
-is executed before that function returns. This is why the idiom is to use the defer
-keyword right after the With call, as you see on line 26. Not doing this will cause
-memory leaks in your program.
+Крайне важно, чтобы любая функция отмены, возвращаемая функцией `With`, была выполнена до возврата функции.
```
Comment (shalimski) [inline _content/tour/rus/context.article:282]:
```suggestion
Крайне важно, чтобы любая функция отмены, возвращаемая функцией `With`, выполнялась до того, как эта функция вернет результат.
```

File: _content/tour/rus/context.article [hunk 19]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
...
-It’s critically important that any cancel function returned from a With function
-is executed before that function returns. This is why the idiom is to use the defer
-keyword right after the With call, as you see on line 26. Not doing this will cause
-memory leaks in your program.
+Крайне важно, чтобы любая функция отмены, возвращаемая функцией `With`, была выполнена до возврата функции. 
+Поэтому ключевое слово defer следует использовать сразу после вызова `With`, как показано в строке 26. 
+Невыполнение этого требования приведет к утечке памяти в вашей программе.
 
-*When* *a* *Context* *is* *canceled,* *all* *Contexts* *derived* *from* *it* *are* *also* *canceled*
+*Когда* *контекст* *удаляется*, *все* *производные* *от* *него* *контексты* *также* *удаляются*.
```
Comment (shalimski) [inline _content/tour/rus/context.article:286]:
```suggestion
*Когда* *контекст* *отменяется*, *все* *производные* *от* *него* *контексты* *также* *отменяются*.
```

File: _content/tour/rus/context.article [hunk 20]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
...
 
-The use of value semantics for the Context API means each new Context value is
-given everything the parent Context has plus any new changes. This means if a
-parent Context is cancelled, all children derived by that parent Context are
-cancelled as well.
+Использование семантики значений для `Context` API означает, что каждому новому значению `Context` присваивается все, 
+что есть у родительского `Context`, плюс все новые модификации. Это означает, что если родительский Контекст удаляется,
```
Comment (shalimski) [inline _content/tour/rus/context.article:289]:
```suggestion
что есть у родительского `Context`, плюс любые новые изменения. Это означает, что если родительский Контекст отменяется, 
```

File: _content/tour/rus/context.article [hunk 21]
```diff
@@ -277,28 +270,24 @@ value that is then propagated forward.
...
-*When* *a* *Context* *is* *canceled,* *all* *Contexts* *derived* *from* *it* *are* *also* *canceled*
+*Когда* *контекст* *удаляется*, *все* *производные* *от* *него* *контексты* *также* *удаляются*.
 
-The use of value semantics for the Context API means each new Context value is
-given everything the parent Context has plus any new changes. This means if a
-parent Context is cancelled, all children derived by that parent Context are
-cancelled as well.
+Использование семантики значений для `Context` API означает, что каждому новому значению `Context` присваивается все, 
+что есть у родительского `Context`, плюс все новые модификации. Это означает, что если родительский Контекст удаляется, 
+то удаляются и все дочерние Контексты, производные от этого родительского Контекста.
```
Comment (shalimski) [inline _content/tour/rus/context.article:290]:
```suggestion
то отменяются и все дочерние Контексты, производные от этого родительского Контекста.
```

File: _content/tour/rus/context.article [hunk 22]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
 
-This program creates a Context value that can be cancelled on line 04. Then on
-lines 13-25, ten goroutines are created. Each goroutine places their unique id
-inside their own Context value on line 19. The call to WithValue is passed the
-main function’s Context value as its parent. Then on line 22, each goroutine waits
-until their Context is cancelled.
+Эта программа создает значение Context, которое может быть удалено в строке 04. Затем в строках 13-25 создаются десять горутин. 
+Каждая горутина помещает свой уникальный ID в свое собственное значение `Context` в строке 19. Вызову `WithValue` передается 
+значение `Context` главной функции в качестве родительского. Затем, в строке 22, каждая горутина ждет, пока ее `Context` 
+не будет очищен.
```
Comment (shalimski) [inline _content/tour/rus/context.article:326]:
```suggestion
не будет отменен.
```

File: _content/tour/rus/context.article [hunk 23]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
 
-On line 28, the main goroutine cancels its Context value and then waits on line 29
-for all ten of the goroutines to receive the signal before shutting down the program.
-Once the cancel function is called, all ten goroutines on line 41 will become
-unblocked and print that they have been cancelled. One call to cancel to cancel
-them all.
+В строке 28 главная горутина очищает свое значение `Context`, а затем в строке 29 ждет, пока все десять горутин получат сигнал,
```
Comment (shalimski) [inline _content/tour/rus/context.article:328]:
```suggestion
В строке 28 главная горутина отменяет свой  `Context`, а затем в строке 29 ждет, пока все десять горутин получат сигнал,
```

File: _content/tour/rus/context.article [hunk 24]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
+Один контекст безопасен для одновременного использования несколькими горутинами.
 
-Do not pass a nil Context, even if a function permits it. Pass a TODO context if
-you are unsure about which Context to use. One of my favorite parts of the Context
-package is the TODO function. I am a firm believer that a programmer is always
-drafting code. This is no different than a writer who is drafting versions of an
-article. You never know everything as you write code, but hopefully you know enough
-to move things along. In the end, you are constantly learning, refactoring and
-testing along the way.
+Не передавайте контекст `nil`, даже если функция это позволяет. Передавайте контекст `TODO`, если вы не уверены,
```
Comment (shalimski) [inline _content/tour/rus/context.article:335]:
```suggestion
Не передавайте в контекст значение `nil`, даже если функция это позволяет. Передавайте контекст `TODO`, если вы не уверены, 
```

File: _content/tour/rus/context.article [hunk 25]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
-you are unsure about which Context to use. One of my favorite parts of the Context
-package is the TODO function. I am a firm believer that a programmer is always
-drafting code. This is no different than a writer who is drafting versions of an
-article. You never know everything as you write code, but hopefully you know enough
-to move things along. In the end, you are constantly learning, refactoring and
-testing along the way.
+Не передавайте контекст `nil`, даже если функция это позволяет. Передавайте контекст `TODO`, если вы не уверены, 
+какой контекст использовать. Одна из моих любимых частей пакета `Context` - функция `TODO`. 
+Я твердо верю, что программист всегда пишет код. Это ничем не отличается от писателя, который набрасывает варианты статьи. 
+Вы никогда не знаете всего, пока пишете код, но, надеюсь, вы знаете достаточно, чтобы двигаться вперед.
```
Comment (shalimski) [inline _content/tour/rus/context.article:338]:
```suggestion
Вы никогда не знаете всего, пока пишете код, но, надеюсь, вы знаете достаточно, чтобы продвигаться вперед. 
```

File: _content/tour/rus/context.article [hunk 26]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
-so using the Background function was out of the question. I needed a temporary
-top-level Context until I figured out where the actual Context was coming from.
-This is when you should use the TODO function over the Background function.
+Было много случаев, когда я знал, что мне нужен контекст, но не был уверен, откуда он возьмется. 
+Я знал, что не отвечаю за создание контекста верхнего уровня, поэтому об использовании функции `Background` не могло быть и речи. 
+Мне нужен был временный контекст верхнего уровня, пока я не узнаю, откуда возьмется настоящий контекст. 
+Именно в таких случаях следует использовать функцию `TODO`, а не функцию `Background`.
 
-*Use* *Context* *values* *only* *for* *request-scoped* *data*
+*Используйте* *значения* *с* *Context* *только* *в* *области* *действия* *запроса* *данных*.
```
Comment (shalimski) [inline _content/tour/rus/context.article:346]:
```suggestion
*Используйте* *значения* *Context* *только* *для* *данных,* *относящихся* *к* *области* *запроса.*
```

File: _content/tour/rus/context.article [hunk 27]
```diff
@@ -331,58 +320,48 @@ cancelled as well.
...
 
-A classic example of the misuse of passing data into a function call using Context
-is with database connections. As a general rule, you want to follow this order when
-moving data around your program.
+Классическим примером неправильной передачи данных при вызове функции с использованием Context являются соединения с базами данных. 
+Как правило, при передаче данных через программу следует придерживаться следующего порядка.
 
-Pass the data as a function parameter This is the clearest way to move data around
-the program without hiding it.
+Передавайте данные как параметр функции Это самый очевидный способ перемещения данных в программу, не скрывая их.
```
Comment (shalimski) [inline _content/tour/rus/context.article:357]:
```suggestion
Передавайте данные как параметр функции. Это самый очевидный способ перемещения данных в программу, не скрывая их.
```

File: _content/tour/rus/context.article [hunk 28]
```diff
@@ -427,13 +405,12 @@ layer to function with an empty Context value.
...
     32     app.Handle("GET", "/v1/users", u.List)
 
-This code constructs a User value and then binds the List method into the route.
-Once again, since the signature of a handler function is unchangeable, using a
-receiver and methods is the next best way to pass data without it being hidden.
+Этот код создает значение User, а затем связывает метод `List` с маршрутом. Опять же, поскольку сигнатура функции-обработчика 
+неизменяема, использование приемника и методов - лучший способ передачи данных без их сокрытия.
 
-*Debugging* *or* *tracing* *data* *is* *safe* *to* *pass* *in* *a* *Context*
+*Отладочные* *или* *трассировочные* *данные* *безопасно* *передавать* *с* *Context*.
```
Comment (shalimski) [inline _content/tour/rus/context.article:411]:
```suggestion
*Отладочные* *или* *трассировочные* *данные* *безопасно* *передавать* *через* *Context*.
```

File: _content/tour/rus/context.article [hunk 29]
```diff
@@ -427,13 +405,12 @@ layer to function with an empty Context value.
...
 
-Data that can be stored and received from a Context value is debug and tracing information.
+Данные, которые могут быть сохранены и получены значением Context, являются отладочной и трассировочной информацией.
```
Comment (shalimski) [inline _content/tour/rus/context.article:413]:
```suggestion
Данные, которые могут быть сохранены и получены из контекста, могут являтся отладочной и трассировочной информацией.
```

File: _content/tour/rus/context.article [hunk 30]
```diff
@@ -462,8 +438,8 @@ information for the request. This information is gathered as the request progres
     89     }
     90     ctx = context.WithValue(ctx, KeyValues, &v)
 
-See how the Values type is constructed on line 86 and then stored inside the Context
-on line 90. It’s the logging middleware that needs most of this information.
+Посмотрите, как тип Values создается в строке 86, а затем хранится в Context в строке 90.
```
Comment (shalimski) [inline _content/tour/rus/context.article:441]:
```suggestion
Посмотрите, как тип Values создается в строке 86, а затем сохраняется внутри Context в строке 90. 
```

File: _content/tour/rus/context.article [hunk 31]
```diff
@@ -481,44 +457,46 @@ on line 90. It’s the logging middleware that needs most of this information.
     37         r.RemoteAddr, time.Since(v.Now),
     38     )
 
-The consequence of passing information through the Context is shown in the code on
-lines 27-30. The code is attempting to retrieve the Values data from the Context
-and checking if the data was there. If the data is not there, then a major integrity
-issue exists and the service needs to shutdown. This is done in the service code by
-sending a special error value back up through the application.
+Последствия передачи информации через Context показаны в коде на
```
Comment (shalimski) [inline _content/tour/rus/context.article:460]:
```suggestion
Результат передачи информации через контекст  показан в коде на
```

File: _content/tour/rus/context.article [hunk 32]
```diff
@@ -481,44 +457,46 @@ on line 90. It’s the logging middleware that needs most of this information.
...
-- Replace a Context using WithCancel, WithDeadline, WithTimeout, or WithValue.
-- When a Context is canceled, all Contexts derived from it are also canceled.
-- Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.
-- Do not pass a nil Context, even if a function permits it. Pass context.TODO if you are unsure about which Context to use.
-- Use context Values only for request-scoped data that transits processes and APIs, not for passing optional parameters to functions.
-- The same Context may be passed to functions running in different goroutines; Contexts are safe for simultaneous use by multiple goroutines.
+- Входящие запросы к серверу должны создавать Context.
+- Исходящие вызовы к серверам должны принимать Context. 
+- Цепочка вызовов функций между ними должна распространять Context.
+- Замените Context с помощью `WithCancel`, `WithDeadline`, `WithTimeout` или `WithValue`.
```
Comment (shalimski) [inline _content/tour/rus/context.article:477]:
```suggestion
- Заменяйте Context с помощью `WithCancel`, `WithDeadline`, `WithTimeout` или `WithValue`.
```

File: _content/tour/rus/context.article [hunk 33]
```diff
@@ -481,44 +457,46 @@ on line 90. It’s the logging middleware that needs most of this information.
...
+- Когда Context отменяется, все производные от него `Context` также отменяется.
+- Не храните `Context` внутри  `struct`; вместо этого передавайте `Context` явно каждой функции, которой он нужен.
```
Comment (shalimski) [inline _content/tour/rus/context.article:479]:
```suggestion
- Не храните `Context` внутри структур, вместо этого передавайте `Context` явно каждой функции, которой он нужен.
```

File: _content/tour/rus/context.article [hunk 34]
```diff
@@ -481,44 +457,46 @@ on line 90. It’s the logging middleware that needs most of this information.
...
+- Не передавайте Context с нулевым значением, даже если функция это допускает. Передавайте `context.TODO`, если вы не уверены в том, какой Context использовать.
```
Comment (shalimski) [inline _content/tour/rus/context.article:480]:
```suggestion
- Не передавайте вместо Context значение `nil`, даже если функция это допускает. Передавайте `context.TODO`, если вы не уверены в том, какой Context использовать.
```

**Review by shalimski — CHANGES_REQUESTED** [review]

**Pull request merged**